tensorflow/contrib/cmake/CMakeLists.txt

```diff
@@ -153,13 +156,25 @@ endif()
 
 # MSVC SIMD instructions
 if (tensorflow_WIN_CPU_SIMD_OPTIONS)
+  include(CheckCXXCompilerFlag)
+  add_definitions(-DINTEL_MKL -DEIGEN_USE_VML)
+  if (NOT tensorflow_ENABLE_MKLDNN_SUPPORT)
+    add_definitions(-DINTEL_MKL_ML)
+  endif()
+  CHECK_CXX_COMPILER_FLAG("-fopenmp" COMPILER_OPT_OPENMP_SUPPORT)
+  if (COMPILER_OPT_OPENMP_SUPPORT)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
+  endif()
   if (WIN32)
-    CHECK_CXX_COMPILER_FLAG("${tensorflow_WIN_CPU_SIMD_OPTIONS}" COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
-    if(COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${tensorflow_WIN_CPU_SIMD_OPTIONS}")
-    else()
-      message(FATAL_ERROR "${tensorflow_WIN_CPU_SIMD_OPTIONS} not supported")
-    endif()
+    set(WIN_SIMD_FLAGS "/arch:AVX2" "/arch:AVX" "/arch:SSE2" "/arch:SSE")
```

Comment:
This ignores whatever flags are set in `tensorflow_WIN_CPU_SIMD_OPTIONS`. I think this will cause problems for the existing builds that only attempt to set `/arch:AVX`.

Reply:
This is correct. We have to have the ability to build with or without SIMD options.
Please revert this to make sure we control the optimization level with `tensorflow_WIN_CPU_SIMD_OPTIONS`

Moreover:
Visual studio, by default generates SSE code unless forced otherwise:
https://msdn.microsoft.com/en-us/library/7t5yh4fd.aspx
And enabling AVX2 also enables AVX:
https://msdn.microsoft.com/en-us/library/7t5yh4fd.aspx
https://msdn.microsoft.com/en-us/library/b0084kay.aspx

Reply:
Hi @gunan, thanks for your reply.
I make a new commit to keep the meaning of `tensorflow_WIN_CPU_SIMD_OPTIONS` unchanged. And add an new option `tensorflow_ENABLE_MKL_SUPPORT` to include MKL